### PR TITLE
feat(New Features) : Revamp PA & FI API code snippets to support new features

### PR DESCRIPTION
**Pull request by vamshimupparaju**
As per new features in PA and FI end points added those new features to code snippets.

### CODE REVIEW

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineSingleUnitExample.cs [hunk 1]
```diff
@@ -142,12 +144,23 @@ private static PACalculationParametersRoot GetPaCalculationParameters()
             var paComponentId = componentsResponse.Data.FirstOrDefault(component => (component.Value.Name == ComponentName && component.Value.Category == ComponentCategory)).Key;
             Console.WriteLine($"PA Component Id : {paComponentId}");
 
+            var pricingSourcesApi = new PricingSourcesApi(GetApiConfiguration());
+
+            var pricingSourcesResponse = pricingSourcesApi.GetPAPricingSources(PricingSourceName, PricingSourceCategory, PricingSourceDirectory);
+
+            var paPricingSourceId = pricingSourcesResponse.Data.FirstOrDefault(pricingSource => (pricingSource.Value.Name == PricingSourceName && pricingSource.Value.Category == PricingSourceCategory && pricingSource.Value.Directory == PricingSourceDirectory)).Key;
+            Console.WriteLine($"PA PricingSource Id : {paPricingSourceId}");
```
[the remark this answers is not in the document]
Comment (vamshimupparaju) [reply examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineSingleUnitExample.cs]: Updated

**Review by AkankshaAcharya — APPROVED** [review]

**Review by ksreeramoj — APPROVED** [review]

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitExample.cs [hunk 2]
```diff
@@ -133,12 +136,23 @@ private static PACalculationParametersRoot GetPaCalculationParameters()
...
             var paAccounts = new List<PAIdentifier> { paAccountIdentifier };
             var paBenchmarkIdentifier = new PAIdentifier(Benchmark);
             var paBenchmarks = new List<PAIdentifier> { paBenchmarkIdentifier };
 
-            var paCalculation = new PACalculationParameters(paComponentId, paAccounts, paBenchmarks);
+            var paPortfolioPricingSources = new List<PACalculationPricingSource> { new PACalculationPricingSource(id: paPricingSourceId) };
+
+            var paDataSources = new PACalculationDataSources(portfoliopricingsources: paPortfolioPricingSources, useportfoliopricingsourcesforbenchmark: true);
+
+            var paCalculation = new PACalculationParameters(paComponentId, paAccounts, paBenchmarks, datasources: paDataSources);
```
Comment (saigiridhar21) [inline examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineMultipleUnitExample.cs:155]: Can you please pass all of them as named arguments?

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineSingleUnitExample.cs [hunk 3]
```diff
@@ -142,12 +144,23 @@ private static PACalculationParametersRoot GetPaCalculationParameters()
...
             var paAccounts = new List<PAIdentifier> { paAccountIdentifier };
             var paBenchmarkIdentifier = new PAIdentifier(Benchmark);
             var paBenchmarks = new List<PAIdentifier> { paBenchmarkIdentifier };
 
-            var paCalculation = new PACalculationParameters(paComponentId, paAccounts, paBenchmarks);
+            var paPortfolioPricingSources = new List<PACalculationPricingSource> { new PACalculationPricingSource(id: paPricingSourceId) };
+
+            var paDataSources = new PACalculationDataSources(portfoliopricingsources: paPortfolioPricingSources, useportfoliopricingsourcesforbenchmark:true); 
+
+            var paCalculation = new PACalculationParameters(paComponentId, paAccounts, paBenchmarks, datasources: paDataSources);
```
Comment (saigiridhar21) [inline examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineSingleUnitExample.cs:163]: Can you please pass all of them as named arguments?

Comment (vamshimupparaju) [reply examples/FactSet.AnalyticsAPI.Engines.Example/Examples/PAEngineSingleUnitExample.cs]: Updated

File: examples/FactSet.AnalyticsAPI.Engines.Example/Examples/FIEngineExample.cs [hunk 4]
```diff
@@ -65,7 +79,17 @@ public static void Main(string[] args)
...
+                {
+                    IgnoreSinkingFund = true
+                };
+
+                var fiMunicipalBondsForJobSettings = new FIMunicipalBondsForJobSettings
+                {
+                    IgnoreSinkingFund = true
+                };
+
+                var fiJobSettings = new FIJobSettings("20201201", marketEnvironment: ratePath, bankLoans: fiBankLoansForJobSettings, municipalBonds: fiMunicipalBondsForJobSettings);
```
Comment (saigiridhar21) [inline examples/FactSet.AnalyticsAPI.Engines.Example/Examples/FIEngineExample.cs:92]: Please use named arguments.

Comment (vamshimupparaju) [reply examples/FactSet.AnalyticsAPI.Engines.Example/Examples/FIEngineExample.cs]: Updated

**Review by saigiridhar21 — APPROVED** [review]

**Pull request merged**